### PR TITLE
Bumping clippy & fixing warnings

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: ethcore/src/pod_account.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ impl From<ethjson::spec::Account> for PodAccount {
 		PodAccount {
 			balance: a.balance.map_or_else(U256::zero, Into::into),
 			nonce: a.nonce.map_or_else(U256::zero, Into::into),
-			code: a.code.map(Into::into).or(Some(Vec::new())),
+			code: a.code.map(Into::into).or_else(|| Some(Vec::new())),
```
Comment (keorn) [inline ethcore/src/pod_account.rs]: interesting, would be nicer if `or` would be lazy...

File: Cargo.lock [hunk 2]
```diff
@@ -1517,3 +1517,147 @@ dependencies = [
  "xml-rs 0.3.0 (registry+https://github.com/rust-lang/crates.io-index)",
 ]
 
+[metadata]
```
Comment (arkpar) [inline Cargo.lock]: Is this backwards compatible with stable?

File: util/src/overlaydb.rs [hunk 3]
```diff
@@ -192,6 +192,7 @@ impl HashDB for OverlayDB {
 }
 
 #[test]
+#[cfg_attr(feature="dev", allow(blacklisted_name))]
```
Comment (arkpar) [inline util/src/overlaydb.rs]: just curious, why is it blacklisted?

File: ethcore/src/pod_account.rs (hunk 1 above)
Comment (arkpar) [inline ethcore/src/pod_account.rs]: `or_else` is the lazy version of `or`. Creating an empty vector on stack costs almost nothing though.

File: Cargo.lock (hunk 2 above)
Comment (tomusdrw) [inline Cargo.lock]: It seems so, build has passed.

File: util/src/overlaydb.rs (hunk 3 above)
Comment (tomusdrw) [inline util/src/overlaydb.rs]: So the function is using names like `foo` and `bar` which are blacklisted by default: https://github.com/Manishearth/rust-clippy/wiki#blacklisted_name

I could rewrite the test to use some different names, but not sure if it's worth it. If it happens in the future I will disable this lint completely.

**Pull request merged**